File: algorithms/sorting/selection-sort/selection-sort.mjs
/**
 * Return a sorted copy of the input using Selection Sort.
 * The original array is never mutated.
 */
export function selectionSort(numbers) {
  const array = [...numbers];

  for (let boundary = 0; boundary < array.length - 1; boundary += 1) {
    let minIndex = boundary;

    for (let index = boundary + 1; index < array.length; index += 1) {
      if (array[index] < array[minIndex]) minIndex = index;
    }

    if (minIndex !== boundary) {
      [array[boundary], array[minIndex]] = [array[minIndex], array[boundary]];
    }
  }

  return array;
}

File: algorithms/sorting/selection-sort/selection_sort.py
def selection_sort(numbers):
    """Return a sorted copy of numbers using Selection Sort."""
    array = numbers.copy()

    for boundary in range(len(array) - 1):
        min_index = boundary

        for index in range(boundary + 1, len(array)):
            if array[index] < array[min_index]:
                min_index = index

        if min_index != boundary:
            array[boundary], array[min_index] = array[min_index], array[boundary]

    return array
